src/DGtal/base/CowPtr.h

```diff
@@ -75,9 +75,13 @@ namespace DGtal
     // no need for ~CowPtr - the CountedPtr takes care of everything.
     CowPtr(const CowPtr& r) noexcept             : myPtr(r.myPtr) {}
     /**
-     * @todo JOL: check this.
+     * Builds a copy-on-write pointer from a counted pointer. Requires
+     * an extra dummy parameter in order to solve ambiguities when
+     * casting from Clone<T> to CowPtr<T>.
```

Comment:
Don't you mean from `CounterPtr` to `CowPtr` ? (but I didn't find the ambiguities you talk about)

Reply:
Before there was a constructor CowPtr<T>::CountedPtr<t>( p ). And a conversion operator Clone<T>::operator CountedPtr<T>(). So I could not write an operator Clone<T>::operator CowPtr<T>() because the compiler was complaining about ambiguities when building a CowPtr<T> from a Clone<T>.
By adding an extra parameter in the constructor CowPtr<T>::CountedPtr<t>( p, dummy ), there is no more an ambiguity and I was able to write the two conversion operator in Clone<T>.
Should I say more in the doc ?

Reply:
Ok I see. I have to check that a little bit more since I found it strange that we introduce such an ambiguity (but I concede that it may be necessary). Does both way (`Clone -> CountedPtr -> CowPtr` and  `Clone -> CowPtr`) lead to the same result?

Reply:
Ok, thanks for the additional doc!